cognite/client/data_classes/data_modeling/typed_instances.py

```diff
@@ -230,17 +230,33 @@ def _load_properties(
     cls: type, resource: dict[str, Any], instance_properties: frozenset[str], signature: inspect.Signature
 ) -> dict[str, Any]:
     output: dict[str, Any] = {}
+    to_search = [cls]
+    property_by_name: dict[str, PropertyOptions] = {}
+    while to_search:
+        current_cls = to_search.pop()
+        for name, value in current_cls.__dict__.items():
+            if isinstance(value, PropertyOptions):
+                property_by_name[name] = value
+        to_search.extend(
+            [
+                b
+                for b in current_cls.__bases__
+                if {b not in {object, TypedNode, TypedEdge, Edge, Node, TypedNodeApply, TypedEdgeApply}}
+            ]
+        )
```

Comment:
Let's add unit tests to `tests_unit/test_data_classes/test_data_models/test_typed_instances.py` for these cases.